src/System.Runtime/tests/System/Reflection/ModuleTests.cs

```diff
@@ -114,12 +109,6 @@ public void TestGetType(Type type)
         }
 
         [Fact]
-        public void TestToString()
-        {
-            Assert.Equal("System.Runtime.Tests.dll", Module.ToString());
```

Comment:
Module.Name and Module.FQN are tied to the filename that the assembly is stored in. This is a desktop/CoreCLR (and partly OS)-centric concept that has no equivalent on AOT where the are no files for individual assemblies. 

Unless we can come up with a different model for module names, my approach is to treat those things as diagnostic information only in the post-multimodule world - i.e. don't test their contents.

The same approach for ToString() for many types (not all of them.)
 
GetCallingAssembly() is untrustworthy as it's tied to inlining decisions made by compiler technology (or demands expensive work to make it not tied.) I would again treat these as "if available, use to help in debugging and take your chances, but don't treat their results as contractual."